### ISSUE TITLE
Very high query count deleting pages when using PostgreSQL search

### ISSUE THREAD

**Issue by kaedroho** [wagtail/wagtail#4649]
I found on a site that the following query is run 688 times on average when pages are deleted:

```
DELETE FROM "postgres_search_indexentry" WHERE ("postgres_search_indexentry"."content_type_id" = %s AND "postgres_search_indexentry"."object_id" IN (%s))
```

Wagtail version: 2.0
PostgreSQL version: 9.6.8

**Comment by BertrandBordage** [comment]
This was fixed by 2bf3110ea112f0d479ef5befe92adebdee929b6f. Maybe we should backport it in 2.0.x?

**Comment by kaedroho** [comment]
Ahh, great! Isn't a major issue and it's easy to upgrade from 2.0 -> 2.1 so I'd leave it.
